arrows/vxl/average_frames.cxx

```diff
@@ -0,0 +1,722 @@
+  d->exp_weight = config->get_value< double >( "exp_weight" );
+  d->round = config->get_value< bool >( "round" );
+  d->output_variance = config->get_value< bool >( "output_variance" );
+}
+
+bool
+average_frames
+::check_configuration( vital::config_block_sptr config ) const
+{
+  return true;
+}
+
+// Perform stitch operation
+kwiver::vital::image_container_sptr
+average_frames
+::filter( kwiver::vital::image_container_sptr image_data )
+{
+  // Perform Basic Validation
+  if( !image_data )
+  {
+    return image_data;
+  }
+
+  // Get input image
+  vil_image_view_base_sptr view =
+    vxl::image_container::vital_to_vxl( image_data->get_image() );
+
+  // Perform different actions based on input type
+#define HANDLE_CASE(T)                                                 \
+  case T:                                                              \
+    {                                                                  \
+      typedef vil_pixel_format_type_of<T >::component_type pix_t;      \
+      vil_image_view< pix_t > uncast_input = view;                     \
+      vil_image_view< double > input;                                  \
+      vil_convert_cast( uncast_input, input );                         \
+                                                                       \
+      d->load_model( false );                                          \
+                                                                       \
+      if( !d->output_variance )                                        \
+      {                                                                \
+        vil_image_view< double > output;                               \
+        d->float_averager->process_frame( input, output );             \
+        return std::make_shared< vxl::image_container >( output );     \
+      }                                                                \
+      else                                                             \
+      {                                                                \
+        vil_image_view< double > tmp, output;                          \
+        d->float_averager->process_frame( input, tmp, output );        \
+        return std::make_shared< vxl::image_container >( output );     \
+      }                                                                \
+    }                                                                  \
+    break;                                                             \
+
+  switch (view->pixel_format())
+  {
+    HANDLE_CASE(VIL_PIXEL_FORMAT_BOOL);
+    HANDLE_CASE(VIL_PIXEL_FORMAT_SBYTE);
+    HANDLE_CASE(VIL_PIXEL_FORMAT_UINT_16);
+    HANDLE_CASE(VIL_PIXEL_FORMAT_INT_16);
+    HANDLE_CASE(VIL_PIXEL_FORMAT_UINT_32);
+    HANDLE_CASE(VIL_PIXEL_FORMAT_INT_32);
+    HANDLE_CASE(VIL_PIXEL_FORMAT_UINT_64);
+    HANDLE_CASE(VIL_PIXEL_FORMAT_INT_64);
+    HANDLE_CASE(VIL_PIXEL_FORMAT_FLOAT);
+    HANDLE_CASE(VIL_PIXEL_FORMAT_DOUBLE);
+#undef HANDLE_CASE
+
+  default:
```

Comment:
This is bad practice to just assume the default case is a `vxl_byte` type.  It's possible that we could be missing a type above or someone could add a new type in the future that is not handle above and the default will then incorrectly assume it is a byte.  There is no reason not to use `case VIL_PIXEL_FORMAT_BYTE:` instead.  The `default` case should probably log an error about an unknown type and return the null image.

Also, I'm looking at this out of context, but it seems very odd that the byte image case returns a byte image output if `!d->output_variance` but otherwise (and for all other types) returns a double image.